Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -69,6 +69,13 @@
 @interface CBLWebSocket () <NSStreamDelegate>
 @end
 
+// For controlling async closing, closed states. 
+typedef enum {
+    kCBLWebsocketStateClosed = 0,       ///< The websocket was closed.
+    kCBLWebsocketStateClosing,          ///< The websocket was asked to close but in progress.
+    kCBLWebsocketStateConnecting,       ///< The websocket is created and requested to connect.
+} CBLWebsocketState;
```

Comment:
It feels uncompleted as it jumps from Connecting to Closing. Either we add Connected for the sake of completeness OR we change it to just boolean (e.g. closeRequested).

Reply:
I added connected, created, etc. but there is nothing much of a use though. 

Keeping boolean, _closing i will try and update the PR soon

Reply:
After thought, we may not need a flag / state at all. What we need are:

1. Handle the cookie error and call closeWithError:
2. In closeWithError, if already disconnected, just return so that we will not acknowledge closed twice.